From: Stewart Hildebrand <stewart.hildebrand@dornerworks.com>
Date: Fri, 2 Aug 2019 23:05:20 -0400
Subject: [PATCH 4/4] Add RPi4 (bcm2711) device tree for Xen (32-bit Linux)

Boot memory layout:
0x00200000 - 0x003fffff xen (max size 2MiB)
0x00400000 - 0x013fffff Image (max size 16MiB)

For the pcie_0 device, add device_type = "pci" so that Xen knows to map
the PCI address space.
---
 arch/arm/boot/dts/Makefile                |  1 +
 arch/arm/boot/dts/bcm2711-rpi-4-b-xen.dts | 19 +++++++++++++++++++
 2 files changed, 20 insertions(+)
 create mode 100644 arch/arm/boot/dts/bcm2711-rpi-4-b-xen.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index c75b6f549b46..d5a7ef831fbb 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -9,6 +9,7 @@ dtb-$(CONFIG_ARCH_BCM2835) += \
 	bcm2710-rpi-2-b.dtb \
 	bcm2710-rpi-3-b.dtb \
 	bcm2711-rpi-4-b.dtb \
+	bcm2711-rpi-4-b-xen.dtb \
 	bcm2710-rpi-3-b-plus.dtb \
 	bcm2710-rpi-cm3.dtb
 
diff --git a/arch/arm/boot/dts/bcm2711-rpi-4-b-xen.dts b/arch/arm/boot/dts/bcm2711-rpi-4-b-xen.dts
new file mode 100644
index 000000000000..b4b8f993810c
--- /dev/null
+++ b/arch/arm/boot/dts/bcm2711-rpi-4-b-xen.dts
@@ -0,0 +1,19 @@
+#include "bcm2711-rpi-4-b.dts"
+
+/ {
+	chosen {
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		xen,xen-bootargs = "console=dtuart dtuart=/soc/serial@7e215040 sync_console dom0_mem=512M dom0_max_vcpus=1 bootscrub=0";
+
+		dom0 {
+			compatible = "xen,linux-zimage", "xen,multiboot-module";
+			reg = <0x00400000 0x01000000>;
+		};
+	};
+};
+
+&pcie_0 {
+	device_type = "pci";
+};
-- 
2.22.0

